### PR TITLE
Feature/webapi project added

### PR DESCRIPTION
**Pull request by gitrepository100**
Added ExceptionMiddleware, please, comment

### CODE REVIEW

File: OneSchedule.Domain.Abstractions/OneSchedule.Domain.Abstractions.csproj [hunk 1]
```diff
@@ -1,11 +1,12 @@
-<Project Sdk="Microsoft.NET.Sdk">
+﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
     <TargetFramework>net5.0</TargetFramework>
   </PropertyGroup>
 
   <ItemGroup>
     <PackageReference Include="MongoDB.Driver" Version="2.13.1" />
+    <PackageReference Include="Telegram.Bot" Version="16.0.2" />
```
Comment (elkavitaly) [inline OneSchedule.Domain.Abstractions/OneSchedule.Domain.Abstractions.csproj]: This package was added to all projects. It should be added only to that project which uses this package

File: OneSchedule.Domain.Abstractions/OneScheduleInternalException.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace OneSchedule.Domain.Abstractions
+{
+    class OneScheduleInternalException
```
Comment (elkavitaly) [inline OneSchedule.Domain.Abstractions/OneScheduleInternalException.cs]: Specify the purpose of exception, not the Project name

File: OneSchedule.Domain.Abstractions/OneScheduleInternalException.cs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        : Exception
```
Comment (elkavitaly) [inline OneSchedule.Domain.Abstractions/OneScheduleInternalException.cs]: Combine 2 rows, the row is short there is no need to split it

File: OneSchedule/ExceptionMiddleware/ErrorDetails.cs [hunk 4]
```diff
@@ -0,0 +1,15 @@
+﻿using System.Text.Json;
+
+namespace OneSchedule
+{
+    public class ErrorDetails
```
Comment (elkavitaly) [inline OneSchedule/ExceptionMiddleware/ErrorDetails.cs]: There are 2 same classes in solution

File: OneSchedule/ExceptionMiddleware/ExceptionMiddleware.cs [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+using System.Net;
+using System.Threading.Tasks;
+
+namespace OneSchedule
+{
+    public class ExceptionMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private readonly ILogger _logger;
+        public ExceptionMiddleware(RequestDelegate next, ILogger logger)
```
Comment (elkavitaly) [inline OneSchedule/ExceptionMiddleware/ExceptionMiddleware.cs]: Add empty line

File: OneSchedule/ExceptionMiddleware/ExceptionMiddleware.cs [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+            _logger = logger;
+            _next = next;
+        }
+        public async Task InvokeAsync(HttpContext httpContext)
+        {
+            try
+            {
+                await _next(httpContext);
+            }
+            catch (Exception ex)
```
Comment (elkavitaly) [inline OneSchedule/ExceptionMiddleware/ExceptionMiddleware.cs]: Add handling for our own exception

File: OneSchedule/ExceptionMiddleware/ExceptionMiddleware.cs [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+            {
+                _logger.LogError($"Something went wrong: {ex}");
+                await HandleExceptionAsync(httpContext, ex);
+            }
+        }
+        private async Task HandleExceptionAsync(HttpContext context, Exception exception)
+        {
+            context.Response.ContentType = "application/json";
+            context.Response.StatusCode = (int)HttpStatusCode.InternalServerError;
+            await context.Response.WriteAsync(new ErrorDetails()
```
Comment (elkavitaly) [inline OneSchedule/ExceptionMiddleware/ExceptionMiddleware.cs]: Have you checked the format of response which Telegram can process. I am not sure that this is a right format

File: OneSchedule/ExceptionMiddleware/ExceptionMiddlewareExtensions.cs [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+using Microsoft.AspNetCore.Diagnostics;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using System.Net;
+
+namespace OneSchedule
+{
+    public static class ExceptionMiddlewareExtensions
+    {
+        public static void ConfigureCustomExceptionMiddleware(this IApplicationBuilder app, ILogger logger)
```
Comment (elkavitaly) [inline OneSchedule/ExceptionMiddleware/ExceptionMiddlewareExtensions.cs]: Check how middleware extension classes should be named

Comment (elkavitaly) [inline OneSchedule/ExceptionMiddleware/ExceptionMiddlewareExtensions.cs]: Use DI directly in your middleware class and don't pass logger as a parameter

File: OneSchedule/OneSchedule.csproj [hunk 9]
```diff
@@ -5,7 +5,9 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <PackageReference Include="Microsoft.Owin.Diagnostics" Version="4.2.0" />
```
Comment (elkavitaly) [inline OneSchedule/OneSchedule.csproj]: What is the purpose of this package?

File: OneSchedule/Startup.cs [hunk 10]
```diff
@@ -32,7 +32,8 @@ public void ConfigureServices(IServiceCollection services)
             });
         }
 
-        public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
+        public void Configure(IApplicationBuilder app, IWebHostEnvironment env, ILogger<Startup> logger)
```
Comment (elkavitaly) [inline OneSchedule/Startup.cs]: Same as in previous comment about middleware

File: .gitignore [hunk 11]
```diff
@@ -63,7 +63,7 @@ project.fragment.lock.json
 artifacts/
 launchSettings.json
 appsettings.*.json
-
+TestController.cs
```
Comment (elkavitaly) [inline .gitignore]: Don't specify some temporary files in gitignore, they shouldn't be attached to git at all

File: OneSchedule.Domain.Abstractions/OneScheduleInternalException.cs [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+        public OneScheduleInternalException(string message)
+            : base(message)
+        {
+        }
+
+        public OneScheduleInternalException(string message, Exception inner)
+            : base(message, inner)
+        {
+        }
+
```
Comment (elkavitaly) [inline OneSchedule.Domain.Abstractions/OneScheduleInternalException.cs]: Empty lines

File: OneSchedule/ExceptionMiddleware/ExceptionMiddleware.cs [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+
+namespace OneSchedule
+{
+    public class ExceptionMiddleware
```
Comment (elkavitaly) [inline OneSchedule/ExceptionMiddleware/ExceptionMiddleware.cs]: Please, rename it to `ExceptionHandlingMiddleware`

**Review by elkavitaly — CHANGES_REQUESTED** [review]

File: OneSchedule/OneSchedule.csproj (hunk 9 above)
Comment (gitrepository100) [reply OneSchedule/OneSchedule.csproj]: +

**Pull request closed without merging**